plugins/jobtype/src/azkaban/jobtype/connectors/TdchParameters.java

```diff
@@ -0,0 +1,349 @@
+        if(!StringUtils.isEmpty(_sourceTdTableName) && !StringUtils.isEmpty(_sourceQuery)) {
+          throw new IllegalArgumentException("Cannot choose multiple source");
+        }
+      }
+
+      if(isHdfsToTd) {
+        _tdchType = TdchType.HDFS_TO_TERADATA;
+      } else {
+        _tdchType = TdchType.TERADATA_TO_HDFS;
+      }
+    }
+  }
+
+  public String[] toTdchParams() {
+    ImmutableList.Builder<String> listBuilder = ImmutableList.builder();
+    if(!StringUtils.isEmpty(_mrParams)) {
+      listBuilder.add(_mrParams);
+    }
+    if(!StringUtils.isEmpty(_libJars)) {
+      listBuilder.add("-libjars").add(_libJars);
+    }
+
+    listBuilder.add("-url")
+               .add(_tdUrl)
+               .add("-classname")
+               .add(_tdJdbcClassName)
+               .add("-fileformat")
+               .add(_fileFormat)
+               .add("-jobtype")
+               .add(_jobType)
+               .add("-username")
+               .add(_userName)
+               .add("-password")
+               .add(_tdPassword)
+               .add("-nummappers")
+               .add(_numMappers);
+
+    if(_avroSchemaPath.isPresent()) {
+      listBuilder.add("-avroschemafile")
+                 .add(_avroSchemaPath.get());
+    }
+
+    if(_fieldSeparator.isPresent()) {
+      listBuilder.add("-separator")
+                 .add(_fieldSeparator.get());
+    }
+
+
+    if(TdchType.HDFS_TO_TERADATA.equals(_tdchType)) {
+      listBuilder.add("-sourcepaths")
+                 .add(_sourceHdfsPath)
+                 .add("-targettable")
+                 .add(_targetTdTableName);
+
+      if(_tdInsertMethod.isPresent()) {
+        listBuilder.add("-method")
+                   .add(_tdInsertMethod.get());
+      }
+    } else if (TdchType.TERADATA_TO_HDFS.equals(_tdchType)){
+      listBuilder.add("-targetpaths")
+                 .add(_targetHdfsPath);
+
+      if(_sourceTdTableName.isPresent()) {
+        listBuilder.add("-sourcetable")
+                   .add(_sourceTdTableName.get());
+
+        if (_tdRetrieveMethod.isPresent()) {
+          listBuilder.add("-method")
+                     .add(_tdRetrieveMethod.get());
+        } else {
+          listBuilder.add("-method")
+                     .add(DEFAULT_RETRIEVE_METHOD);
+        }
+
+      } else if (_sourceQuery.isPresent()) {
+        listBuilder.add("-sourcequery")
+                   .add(_sourceQuery.get());
+      } else {
+        throw new IllegalArgumentException("No source defined."); //This should not happen as it shouldn't have been instantiated by builder.
+      }
+    } else {
+      throw new UnsupportedOperationException("Unsupported TDCH type: " + _tdchType);
+    }
+
+    List<String> paramList = listBuilder.build();
+    String[] params = paramList.toArray(new String[paramList.size()]);
+    return params;
+  }
+
+  @Override
+  public String toString() {
+    return "TdchParameters [_mrParams=" + _mrParams + ", _libJars=" + _libJars + ", _tdJdbcClassName="
```

Comment:
Yep, it's from Eclipse :)
Let me change it to String format also can be done by Eclipse.